### PR TITLE
Restore the behaviour of the app

### PR DESCRIPTION
**Pull request by geffrak**
JIRA: VMR-2289

DESCRIPTION

In order to not break the previous pipelines, the last changes got isolated
inside of hooks. Configs will have to specify that they want to use the new
implementation of the version submission.

The issue was that the app is still used in a Nuke workflow with the
tk-multi-publish2 app and it not be great to break pipelines and change
the expected behaviour on an updates.

Instead, I implemented the new changes inside of hooks and the person adding
the app to a config can choose which implementation to use

### CODE REVIEW

File: hooks/submitter_sgtk.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        }
+
+        if first_frame and last_frame:
+            data["frame_count"] = last_frame - first_frame + 1
+            data["frame_range"] = "%s-%s" % (first_frame, last_frame)
+
+        if sgtk.util.get_published_file_entity_type(self.__app.sgtk) == "PublishedFile":
+            data["published_files"] = sg_publishes
+        else:  # == "TankPublishedFile"
+            if len(sg_publishes) > 0:
```
Comment (jfboismenu) [inline hooks/submitter_sgtk.py]: Why can't other publishes be registered when they are TankPublishedFile ? Where is this limitation coming from?

File: info.yml [hunk 2]
```diff
@@ -64,9 +80,14 @@ configuration:
...
     render_media_hook:
         type: hook
-        description: TODO!!!
+        description: Implements how media get generated while this app is running.
         default_value: '{self}/render_media.py:{self}/{engine_name}/render_media.py'
 
+    submitter_hook:
+        type: hook
+        description: Implements how media get sent to Shotgun
+        default_value: '{self}/submitter_create.py'
```
[the remark this answers is not in the document]
Comment (geffrak) [reply info.yml]: YES. I forgot to put it back after testing. Thank you!

File: hooks/submitter_sgtk.py (hunk 1 above)
Comment (geffrak) [reply hooks/submitter_sgtk.py]: I moved back old code. I don't know the reason behind this limitation.

Comment (jfboismenu) [reply hooks/submitter_sgtk.py]: Bummer....

**Pull request merged**